llvmlite/ir/instructions.py

```diff
@@ -431,9 +431,18 @@ def descr(self, buf):
 
 class LoadInstr(Instruction):
 
-    def __init__(self, parent, ptr, name=''):
-        super(LoadInstr, self).__init__(parent, ptr.type.pointee, "load",
-                                        [ptr], name=name)
+    def __init__(self, parent, ptr, name='', typ=None):
+        if typ is not None:
+            pass
+        elif isinstance(ptr, AllocaInstr):
+            typ = ptr.allocated_type
+        # For compatibility with typed pointers. Eventually this should
+        # probably be removed (when typed pointers are fully removed).
+        elif not ptr.type.is_opaque:
+            typ = ptr.type.pointee
+        else:
+            raise ValueError("Load lacks type.")
```

Comment:
The idea was to keep it somewhat consistent with what's done in `GEPInstr`, but you're right, it might be more confusing than it's worth so I've changed it.

Reply:
Thanks - I hadn't spotted the commonality with `GEPInstr` - I think the simplified form is easier to understand, so thanks for making the change.